ee/modules/030-cloud-provider-openstack/images/cloud-data-discoverer/discoverer.go

```diff
@@ -84,3 +98,252 @@ func (d *Discoverer) InstanceTypes(_ context.Context) ([]v1alpha1.InstanceType,
 
 	return res, nil
 }
+
+func (d *Discoverer) DiscoveryData(ctx context.Context) ([]byte, error) {
+	cctx, cancel := context.WithTimeout(ctx, 10*time.Second)
+	defer cancel()
+
+	obj, err := d.k8sClient.Resource(schema.GroupVersionResource{Group: "", Version: "v1", Resource: "secrets"}).Namespace("kube-system").Get(cctx, "d8-provider-cluster-configuration", metav1.GetOptions{})
```

Comment:
We can use normal (not dynamic) client for secrets. Lets initialize regular go-client and pass to discoverer

Reply:
hm. Why do we need to request secret in OpenstackDiscoverer?
I think, we will do all works with secret in the `Reconciler`.

Reply:
And we do not need to pass kube clienet to discoverer.